### PR TITLE
4.0.8.7 - Minecraft/Spigot 1.17 support

### PR DESCRIPTION
**Pull request by Ghost-chu**

### CODE REVIEW

File: src/main/java/org/maxgamer/quickshop/shop/InventoryPreview.java [hunk 1]
```diff
@@ -111,7 +110,8 @@ public void show() {
             return;
         }
         final int size = 9;
-        inventory = plugin.getServer().createInventory(new QuickShopPreviewInventoryHolder(), size, MsgUtil.getMessage("menu.preview", player));
+
+        inventory = plugin.getServer().createInventory(new QuickShopPreviewInventoryHolder(), 9, MsgUtil.getMessage("menu.preview", player));
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/shop/InventoryPreview.java:114]: What difference?

File: src/main/java/org/maxgamer/quickshop/util/config/ConfigProvider.java [hunk 2]
```diff
@@ -80,9 +80,8 @@ public void reload(boolean defaults) {
         if (config == null) {
             config = new YamlConfiguration();
         }
-        try {
+        try (InputStream defaultConfigStream = plugin.getResource("config.yml")) {
             config.load(configFile);
-            InputStream defaultConfigStream = plugin.getResource("config.yml");
             if (defaultConfigStream != null) {
                 config.setDefaults(YamlConfiguration.loadConfiguration(new InputStreamReader(defaultConfigStream, StandardCharsets.UTF_8)));
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/util/config/ConfigProvider.java:86]: We should also put InputStreamReader to try-with-resources block

File: src/main/java/org/maxgamer/quickshop/util/language/game/MojangGameLanguageImpl.java [hunk 3]
```diff
@@ -69,19 +75,25 @@ public MojangGameLanguageImpl(@NotNull QuickShop plugin, @NotNull String languag
         if ("default".equalsIgnoreCase(languageCode)) {
             Locale locale = Locale.getDefault();
             languageCode = locale.getLanguage() + "_" + locale.getCountry();
+            if (StringUtils.isEmpty(locale.getCountry())) {
+                languageCode = locale.toLanguageTag();
+                plugin.getLogger().info("Your system offer empty data about locale country, we will fallback to system language default country, current language set to " + languageCode + ". If it incorrect, please edit it in config.yml.");
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/util/language/game/MojangGameLanguageImpl.java:80]: What about `locale.getLanguage()` return ""?

**Review by sandtechnology — CHANGES_REQUESTED** [review]

File: src/main/java/org/maxgamer/quickshop/util/language/game/MojangGameLanguageImpl.java (hunk 3 above)
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/util/language/game/MojangGameLanguageImpl.java:80]: so it will broken, but i didn't found better way.

**Pull request merged**